### PR TITLE
Feature/several datasets

### PR DESCRIPTION
**Pull request by GFibrizo**
Loading specific dataset that complies with data requirements from the data owner. This dataset is then linked to a model id (of training request id).

**Notes:**
- This was only tested in IDE with 1 client. Next step is test it with several and with docker compose
- I changed how the test data was being loaded in the model buyer for testing purposes only. I should change it back later.
- When no dataset is loaded because no one complies with the requirements, no message is being logged or sent to the federated trainer/model buyer. This is a **must**.

**EDIT**
- Tested with 5 clients using docker-compose. OK.
- Added temporary fix to how the test dataset is loaded by the model buyer.
- Opened [issue](https://github.com/DeltaML/federated-learning-poc/issues/12#issue-442485089) about last item in the notes

### CODE REVIEW

File: model_buyer/config.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 config = {
-    'server_register_url': "http://cte_federated_trainer:8080/model",
+    'server_register_url': "http://localhost:8080/model",
```
[the remark this answers is not in the document]
Comment (GFibrizo) [reply model_buyer/config.py]: Done.

File: model_buyer/service/model_buyer.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def make_prediction(self, data):
         if not model:
             raise OrderedModelNotFoundException(model_id)
 
-        x_test, y_test = self.data_loader.get_sub_set()
+        x_test, y_test, a,b = self.data_loader.load_random_data()
```
[the remark this answers is not in the document]
Comment (GFibrizo) [reply model_buyer/service/model_buyer.py]: Done

File: federated_trainer/federated_trainer_app.py [hunk 3]
```diff
@@ -78,6 +78,7 @@ def train_model_async():
     logging.info("Initializing async model trainig acording to request {}".format(data))
     logging.info("host {} port {}".format(request.environ['REMOTE_ADDR'], request.environ['REMOTE_PORT']))
     # Validate model type
+    print(data)
```
[the remark this answers is not in the document]
Comment (GFibrizo) [reply federated_trainer/federated_trainer_app.py]: Done

File: federated_trainer/service/data_owner_connector.py [hunk 4]
```diff
@@ -29,16 +29,21 @@ def get_data_owners_model(self, data_owners):
         results = self.async_thread_pool.run(executable=self._get_data_owner_model, args=args)
         return [result for result in results]
 
+    def send_requirements_to_data_owners(self, data_owners, data):
```
Comment (agrojas) [inline federated_trainer/service/data_owner_connector.py]: Esto se puede hacer en paralelo

Comment (GFibrizo) [reply federated_trainer/service/data_owner_connector.py]: Done, I think. Testing with 5 clients with docker-compose finishes ok.

**Pull request merged**